### ISSUE TITLE
LoginProducer.java: At the moment only the "amiha... 

### ISSUE THREAD

**Issue by charlesmike** [self-xdsd/self-web#322]
The puzzle ``1535146848`` originating from #45 has to be resolved:

https://github.com/self-xdsd/self-web/blob/bf574b14bfdedc818f8774694500524750efe2c7/src/main/java/com/selfxdsd/selfweb/LoginProducer.java#45-47

"At the moment only the "amihaiemil" user has the admin role. Everyone who is an admin in the Github or GitLab self-xdsd Organization, should have the admin role.".

Estimation is ``60 minutes``.
If you have any technical questions, don't ask me, I won't be able to help. Open new issues instead.

**Comment by charlesmike** [comment]
@Lumi3011 this is your task now, please go ahead. Deadline (when this ticket should be closed) is ``2021-01-17T10:49:47.635541``.

Estimation here is ``60 minutes``, that's how much you will be paid.

Remember, you don't have to solve everything in this ticket - you can solve it [partially](https://docs.self-xdsd.com/microtasking.html) and leave ``todo`` markers in the code, which will become future tasks.

If you have any questions don't ask me, I'm not a technical person. Open new tickets instead.

If you don't have time or simply don't want to work on this, you can always [resign](https://docs.self-xdsd.com/projectmanager.html#commands).

**Comment by amihaiemil** [comment]
@charlesmike deregister

**Comment by charlesmike** [comment]
> @charlesmike deregister

@amihaiemil ok, I've removed this task from scope. I'm not managing it anymore.

**Comment by nmonokov** [comment]
@amihaiemil hey, i'm not sure how to work with PDD correctly: should i create a new issue or can i drop a comment here? So, i'm confused about this task. I see 2 options here:
1. Whoever has the ROLE_ADMIN should have an admin role in the application.
2. Explicitly who has the admin role in self-xdsd Organization could have that authority. This concept of organizations in VCS is completely new to me and if this is true, I tried to boot your application, to have the full picture of how this Organization part is working. I tried to run jooq scripts from self-storage repository but i constantly fail due to NPE:
`[ERROR] Failed to execute goal org.jooq:jooq-codegen-maven:3.13.2:generate (jooq-codegen) on project self-storage: Error running jOOQ code generation tool: Error generating code for catalog: NullPointerException -> [Help 1]`
I've created schema in MySQL: self_xdsd. 
Updated data in pom with my creds for db:
`<jdbc>
    <driver>com.mysql.cj.jdbc.Driver</driver>
    <url>http://localhost:3306</url>
    <user>root</user>
    <password>root</password>
</jdbc>`

Basically if this task involves something more than just having the ROLE_ADMIN in authority than i'm curious, is there any comprehensive guide on correct application setup?

**Comment by amihaiemil** [comment]
@nmonokov dont do anyrhing about this task yet, please.

I'll come back tomorrow with answers to your questions.

**Comment by amihaiemil** [comment]
@nmonokov 

Oraganizations have nothing to do with git as a VCS. Organizations are just a way for Github and GitLab to group repositories under an "organzation" or "company", instead of a personal user account. Organizations in Github and GitLab allow for better access control, more transparency etc. They also look more professional, if you have a popular product (it looks better if it's owned by an org/company, than by an individual user).

The idea of this task was to allow any user from Github/GitLab, who is an admin on the ``self-xdsd`` organization, to also have admin rights in Self XDSD webapp. However, I realize now this is a bad idea. We might want to let people be admins on the organization, but still not be admins in the Self Web. 

So I think we will close this task, but I'm not sure yet. 

About working for self-xdsd: in principle, the PM has to assign you tasks which are in scope (this task here is not in scope, as you see). At the moment things are a bit slow because we are preparing the first Prod Release, so there is not much going on. I can add you to self-core as well, if you wish, and look for a task for you there.

About running self-web locally: you don't have to run JOOQ. JOOQ generates code from the DB, which is done in self-storage. I'll explain how to set up in self-xdsd/assets, which is a private repo. I'll tag you there.

**Comment by nmonokov** [comment]
@amihaiemil sounds good. I'd like to contribute to self-core as well. At least I'll try :D. If I have any blockers or misunderstandings, is it ok to ping you in the issue thread for advice or direction?

**Comment by amihaiemil** [comment]
@nmonokov sure

**Comment by amihaiemil** [comment]
@nmonokov If you have a GitLab account, I can also add you to self-xdsd/self-rest -- this is a new project, which we develop at GitLab. It will be Self XDSD's RESTful API.

**Comment by nmonokov** [comment]
@amihaiemil yes, i have a gitlab account. The nickname is the same as here.
